component/storage/edv/encryptedformatter.go

```diff
@@ -118,52 +168,72 @@ func (e *EncryptedFormatter) getStructuredDocFromEncryptedDoc(
 	return structuredDoc, nil
 }
 
-// TODO (#2376) Revisit how we're generating EDV document IDs, since it's technically not 100% in line with the spec.
-//  (Spec requires randomly generated IDs)
 // Generates an encrypted document ID based off of key.
-func (e *EncryptedFormatter) formatKey(key string) (string, error) {
-	keyHash, err := e.macCrypto.ComputeMAC([]byte(key))
+func (e *EncryptedFormatter) generateDocumentIDDeterministically(prefix, key string) (string, error) {
```

Comment:
consider update naming to "generateDeterministicDocumentID"

Reply:
I agree, that shorter and better. Done!